tests/message.cpp

```diff
@@ -1,5 +1,9 @@
 #define CATCH_CONFIG_MAIN
+#ifdef USE_EXTERNAL_CATCH2
```

Comment:
This can be avoided as in both cases (catch2, catch1) header name is the same. Instead doing ifdef in the code you can just control include paths from cmake.

Reply:
I think it is better to stick with the include base path a library expects/defines, and I also think the #ifdef is fine in general. However, please confine this to a single point instead of duplicating this across all test files.